src/EnergyPlus/FileSystem.hh

```diff
@@ -85,17 +88,45 @@ namespace FileSystem {
 
     extern std::string const exeExtension;
 
-    enum class InputFileType
+    enum class FileTypes
     {
+        Unknown,
+        // JSON types should go first,
         EpJSON,
-        IDF,
+        GLHE,
+        JSON,
+        last_json_type = JSON,
         CBOR,
         MsgPack,
         UBJSON,
         BSON,
-        Unknown
+        last_binary_json_type = BSON,
+        CSV,
+        TSV,
+        TXT,
+        ESO,
+        MTR,
+        IMF,
+        IDF,
+        last_flat_file_type = IDF
```

Comment:
We add a last enumeration value `Num` so that `FileTypes::Num` can be used to size arrays and as a loop bound.

Reply:
I am unsure how the enumeration values will increment with the features like `last_flat_file_type = IDF`, but it should work the same. That said, I don't envision using this for size arrays or loop bounds unless I implement something like your comment above to use `getEnumerationValue`.

Reply:
`last_flat_file_type = IDF` will continue to work fine. You also don't really need it if you use `BITF_TEST_ANY(BITF(FileType), BITF(TXT) | BITF(MTR) | BITF (IDF) | BITF(IMF))` etc.